WordPress/gutenberg
# Add link color theme_support

## What?

This PR adds theme support for the link color settings.

### **Update: This core patch has been committed https://core.trac.wordpress.org/ticket/58597**

---

Closes https://github.com/WordPress/gutenberg/issues/41857, replaces https://github.com/WordPress/gutenberg/pull/47675
Alternative to https://github.com/WordPress/gutenberg/pull/47451

Related:
Core ticket and patch https://core.trac.wordpress.org/ticket/58597

[core 57460 Backport: appearance-tools theme_support](https://core.trac.wordpress.org/ticket/57460)
[core 56487 Bundled themes: opt-in to appearance tools](https://core.trac.wordpress.org/ticket/56487)

## Why?

There is no way for themes to enable link color settings without theme.json (Theme.json can not be added to all classic themes without friction).

## How?

Adds a conditional check, and if the theme or plugin adds `add_theme_support( 'link-color' );` enables the setting in the block editor.

## Testing Instructions
In your WordPress install, make sure that WP_DEBUG_DISPLAY, WP_DEBUG are set to true.
Activate Twenty Twenty-One.
You should see a _doing it wrong_ message about experimental-link-color.
In your code editor, open functions.php in Twenty Twenty-One.
Around line 333, remove
`add_theme_support( 'experimental-link-color' );`

And add:

```
add_theme_support( 'link-color' );
```
On refresh, the doing it wrong notice should no longer show.

Create a new post.
Add a link to a paragraph block.
Confirm that the paragraph has a link color setting. Select a custom link color and confirm that it works.

_Palette colors will not work on the front, that is a separate issue: 
There are known bugs in Twenty Twenty-One that needs to be solved, you do not need to test them as part of this PR,
the main goal is to confirm that the controls are available._

## Screenshots or screencast
